### DISCUSSION TITLE
NodeJS: Qual ORM usar para banco de dados relacional?

### DISCUSSION THREAD

**Discussion by drianoaz** [frontendbr/forum#1542] [Perguntas e Respostas]
E ae galera, beleza?

Estou iniciando meus estudos sobre NodeJS no backend e, pesquisando sobre o assunto, encontrei diversas alternativas de ORMs para bancos de dados relacionais.

Gostaria de conhecer um pouco mais sobre as ferramentas que vocês estão utilizando:

- Qual ORM você tem utilizado?
- O que levou você a escolhe-lo?
- Qual a curva de aprendizado?
- Como você lida com as migrações/versionamento?
- Como você testa suas APIs que tocam no banco?
  - Você faz mock dos acessos ao banco?
  - Você faz o setup do banco e roda os testes dentro de uma transação dando rollback a cada teste?
  - Você tem alguma dica sobre como manter esse tipo de teste rápido?

**Comment by WillianSugiyama** [comment]
Já utilizei alguns ORM's em node, atualmente estou usando o [TypeORM ](https://typeorm.io/#/)que veio no pacote do NestJS (framework typescript que tenho usado).

Já utilizei o [Sequlize](https://sequelize.org/) também, em questão de performance ambos são bons, a curva de aprendizado de ambos é bem baixa, é bem rápido você os entender desde criar uma entity/model, até fazer as querys. 

Ambos os ORM's citados tem as migrations bem definidas na documentação e isso te ajuda neste quesito.

Em questão de teste, tudo que eu faço são mocks dos acessos, e do que o banco deveria retornar.
